### PR TITLE
Implement Player Class

### PR DESCRIPTION
**Pull request by Soulsteeler123**
Skeleton of Player class

### CODE REVIEW

File: src/AdditionalPylonsModule.cpp [hunk 1]
```diff
@@ -55,27 +53,31 @@ void AdditionalPylonsModule::onUnitEvade(BWAPI::Unit unit) {
...
-
+	if (unit->getPlayer() == BWAPI::Broodwar->self()) {
+		player.onUnitCreate(unit);
+	}
+	else if(unit->getPlayer() != BWAPI::Broodwar->neutral()){
+		enemy.onUnitCreate(unit);
+	}
 }
 
 void AdditionalPylonsModule::onUnitDestroy(BWAPI::Unit unit) {
```
Comment (0x416C657852) [inline src/AdditionalPylonsModule.cpp:83]: 😿onUnitCreate on delete.

**Review by 0x416C657852 — CHANGES_REQUESTED** [review]
lgtm,
just change that thing, yknow the thing from earlier I already forgot about. p sure I commented it somewhere

File: src/Players/Player.cpp [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+//Returns this player's race
+BWAPI::Race Player::returnRace() {
+	return ourRace;
+}
+
+void Player::DisplayInfo(int x, int y) {
+
+	BWAPI::Broodwar->drawTextScreen(x, y, "# of units that %s control: %s", ourRace.toString(), std::to_string(units.size()));
+	BWAPI::Broodwar->drawTextScreen(x, y, "# of units that %s control: %s", ourRace.toString(), std::to_string(buildings.size()));
```
Comment (krogenth) [inline src/Players/Player.cpp]: - `std::to_string(units.size())` should just be `units.size()`, you can use `%d` for outputting integer values
- `ourRace.toString()` won't work, instead, we'll have to check the value stored and create a string off of that(or just output the numerical value stored in the race object)
- the current setup will overlap text on the screen, which we don't want to have happen, preferably, we want as little width used as possible

File: src/Players/Player.cpp [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+void Player::PrintRace(int x, int y) {
+	BWAPI::Broodwar->drawTextScreen(x, y, "This player's race is: %s", ourRace.toString());
```
Comment (krogenth) [inline src/Players/Player.cpp]: This method isn't very useful, since `DisplayInfo` also outputs the race

File: src/Players/Player.cpp [hunk 4]
```diff
@@ -0,0 +1,96 @@
+#include "./Player.h"
+
+//Auto assigns this player's race and sets enemies race to unknown
+void Player::onStart(BWAPI::Race race) {
+	units.clear();
+	buildings.clear();
+	ourRace = race;
+	//enemyRace = BWAPI::Races::Unknown;
```
Comment (krogenth) [inline src/Players/Player.cpp]: remove dead code

File: vs/AdditionalPylons/AdditionalPylons.vcxproj.filters [hunk 5]
```diff
@@ -18,10 +18,16 @@
     <ClCompile Include="..\..\src\AdditionalPylonsModule.cpp">
       <Filter>Source Files</Filter>
     </ClCompile>
+    <ClCompile Include="..\..\src\Players\Player.cpp">
```
Comment (krogenth) [inline vs/AdditionalPylons/AdditionalPylons.vcxproj.filters:30]: Fix vs project filters, they should replicate the storage folder structure

**Review by krogenth — APPROVED** [review]

**Review by 0x416C657852 — APPROVED** [review]

**Pull request merged**